src/Client.ts

```diff
@@ -516,6 +533,116 @@ export default class Client {
+        startTime: nsLongToDate(
+          this.latestInviteNs.sub(INVITE_LOAD_OFFSET * 1000 * 1_000_000)
+        ),
+      },
+      {
+        direction: SortDirection.SORT_DIRECTION_ASCENDING,
+      }
+    )
+    for (const { message, timestampNs } of envelopes) {
+      if (!message || !timestampNs) {
+        console.warn('Invite missing required field')
+        continue
+      }
+      try {
+        const invite = SealedInvitation.fromBytes(
+          b64Decode(message as unknown as string)
+        )
+
+        const serverTime = Long.fromString(timestampNs)
+        const headerTime = invite.v1.header.createdNs
+        if (!headerTime.equals(serverTime)) {
+          console.warn(
+            `Server and header timestamps do not match for invite. Server time: ${serverTime.toString()}. Header time: ${headerTime.toString()}`
+          )
+          continue
+        }
+        // This will decrypt and load the invite, which validates against forged messages
+        await this.saveInvite(invite)
+
+        if (headerTime.gt(this.latestInviteNs)) {
+          this.latestInviteNs = headerTime
+        }
+      } catch (e) {
+        if (!(e instanceof DuplicateTopicError)) {
+          console.warn('Error decoding invite', e)
+        }
+      }
+    }
+  }
+
+  async saveInvite(invite: SealedInvitation) {
+    // Get the invitation (and cache the value) to ensure decryption works
+    // Will throw if the invite cannot be decrypted
+    const unsealed = await invite.v1.getInvitation(this.signedKeys)
+    const header = invite.v1.header
+    const myBundle = this.signedKeys.getPublicKeyBundle()
+    const createdAt = header.createdNs
+    const isSender = header.sender.equals(myBundle)
+    const isRecipient = header.recipient.equals(myBundle)
+    const counterparty = isSender ? header.recipient : header.sender
+
+    if (!isSender && !isRecipient) {
+      throw new Error(
+        'Current account is neither the sender nor the recipient of this bundle'
+      )
+    }
+
+    this.topicKeyManager.addDirectMessageTopic(
+      unsealed.topic,
+      await counterparty.walletSignatureAddress(),
+      {
+        topicKey: {
+          keyMaterial: unsealed.aes256GcmHkdfSha256.keyMaterial,
+          encryptionAlgorithm: EncryptionAlgorithm.AES_256_GCM_HKDF_SHA_256, // TODO: Actually map algorithm instead of hardcoding
+        },
+        allowedSigners: [header.recipient, header.sender],
+        context: unsealed.context,
+        peerAddress: await counterparty.walletSignatureAddress(),
+      },
+      createdAt
+    )
+  }
+
+  async sendInvite(peerAddress: string, context?: InvitationContext) {
+    const recipientContact = await this.getUserContact(peerAddress)
+    if (!recipientContact) {
+      throw new Error(`Recipient with address ${peerAddress} cannot be found`)
+    }
+    const created = new Date()
+    const recipient = SignedPublicKeyBundle.fromLegacyBundle(recipientContact)
```

Comment:
I was planning to use published V2 contact bundle as a trigger for using V2 behavior.